CMakeLists.txt

```diff
@@ -6,3 +6,22 @@
 
 add_subdirectory_ifdef(CONFIG_HAS_STM32CUBE stm32cube)
 add_subdirectory_ifdef(CONFIG_HAS_STM32LIB lib)
+
+# Add BT firmware images upgrade support
+if(CONFIG_BT_SPI)
+  set(hal_dir            ${CMAKE_CURRENT_SOURCE_DIR})
+  set(blob_gen_inc_file  ${ZEPHYR_BINARY_DIR}/include/generated/dtm.bin.inc)
+
+  if(CONFIG_DT_HAS_ST_HCI_SPI_V2_ENABLED)
+    set(hal_blobs_dir      ${hal_dir}/zephyr/blobs/ble-fw/steval-mkboxpro)
+  endif() #CONFIG_DT_HAS_ST_HCI_SPI_V2_ENABLED
+
```

Comment:
Please keep in mind that there lot of others BLE configurations which may be compatible with this combination of Kconfig flags. A feature specific flag would be required at least.

Reply:
Wouldn't CONFIG_DT_HAS_ST_HCI_SPI_V2_ENABLED be enough?

Reply:
Maybe I should check CONFIG_BOARD_SENSORTILE_BOX_PRO...

Reply:
> Wouldn't CONFIG_DT_HAS_ST_HCI_SPI_V2_ENABLED be enough? OK, next week I'll think about it.

I guess there are compatible shields that can come up, right ?
All this is required only because this board is shipped with the wrong binary (at least not up to date). Other boards can be designed with the same chip and be sold with a compatible binary.

Reply:
> > Wouldn't CONFIG_DT_HAS_ST_HCI_SPI_V2_ENABLED be enough? OK, next week I'll think about it.
> 
> I guess there are compatible shields that can come up, right ? All this is required only because this board is shipped with the wrong binary (at least not up to date). Other boards can be designed with the same chip and be sold with a compatible binary.

Not wrong, but a version (which includes host layer) not compatible with zephyr BT stack, which requires the controller-only version. But I agree that I should at least check the board.

Reply:
Board remains to be checked.

Reply:
> Board remains to be checked.

The CMakeLists.txt file has been updated since then. Now I'm checking two boards and setting hal_blobs_dir accordingly:

```
if(CONFIG_BOARD_SENSORTILE_BOX)
  set(hal_blobs_dir      ${hal_dir}/zephyr/blobs/boards/steval-mksbox1v1)
elseif(CONFIG_BOARD_SENSORTILE_BOX_PRO)
  set(hal_blobs_dir      ${hal_dir}/zephyr/blobs/boards/steval-mkboxpro)
else() # it's not a Sensortile Box board
  set(hal_blobs_dir      "")
endif()
```

So, I think it's fine.

Reply:
ok, right.